python/tskit/tables.py

```diff
@@ -1995,6 +2035,19 @@ def __setstate__(self, state):
     @classmethod
     def fromdict(self, tables_dict):
         tables = TableCollection(tables_dict["sequence_length"])
+        # These try-except can be removed once msprime etc are returning dicts with
```

Comment:
I think we'll always have metadata and metadata_schema optional in the dict encoding, so this check will be permanent.

Reply:
https://github.com/tskit-dev/tskit/pull/687/commits/482e1f2b1046a521549e9ea3a9adc1d599fe9aa1